Fix issue #594 related to the fontsize parameter in ``plot``

Tmodrzyk opened the pull request:
The ``fontsize`` parameter in plot has no effect (Issue #594). The fix was very simple, the ``size`` parameter in some matplotlib function calls was overwriting the ``config_matplotlib`` call.

- [x] `python3 -m pytest deepinv/tests` runs successfully.
- [x] `black .` runs successfully.
- [x] `make html` runs successfully (in the `docs/` directory).
- [x] Updated docstrings related to the changes (as applicable).
- [ ] Added an entry to the [CHANGELOG.rst](../CHANGELOG.rst).

samuro95 commented:
Thanks for the correction! Now the config_matplotlib sets the font.size to 17, when it was set to 9 or 12 before. Have you checked that it still looks good ? thanks

samuro95 reviewed (dismissed):
Thanks for the correction! Now the config_matplotlib sets the font.size to 17, when it was set to 9 or 12 before. Have you checked that it still looks good ? thanks 

Moreover, I am wondering if there is a possibility to call config_matplotlib() in the setup.py file so that all the documentation is run with these options, rather than having it in every example file

Tmodrzyk commented:
Hi Sam, thanks for your feedback, I did not think about the documentation, I'll check it. 
As for the ``setup.py`` file you mention I can't seem to find it, where is it located in the library ?

Tmodrzyk commented:
I built the doc locally and checked that everything looks ok. 
Now the **default fontsize is 9** for every ``title`` and ``suptitle`` in the doc's plots.
@Andrewwango @samuro95 could you maybe build the doc too just to double check that I did not break everything ? :smiley:

samuro95 commented on deepinv/utils/plotting.py:
```diff
@@ -255,7 +260,10 @@ def plot(
     """
     # Use the matplotlib config from deepinv
-    config_matplotlib(fontsize=fontsize)
+    if fontsize is not None:
```
Acutally, if we call config_matplotlib every time we call plot, I wonder if we need the config_matplotlib in the conf.py ... do we ?

Tmodrzyk replied:
I think we still need it, otherwise we'd have to either call ``fontsize=9`` in every example in the doc, or set the default fontsize value to 9 (it is currently at 17). 
I find this solution more adapted since we keep the former default value, but also keep the fontsize value related to the doc examples, and don't have to set it to 9 in every example.

samuro95 commented on examples/basics/demo_remote_sensing.py:
```diff
@@ -199,6 +199,7 @@
     compare_no_learning=True,
     no_learning_method="A_dagger",
     show_progress_bar=False,
+    device="cpu",
```
maybe we could add the possibility to run this example on GPU too. Why did you add this line as this is the default device of the trainer ?

Tmodrzyk replied:
This is not the default device of the Trainer when a GPU is available. 
Unfortunately, running the example on GPU broke the example, so that was again a quick fix to make it build.
I fixed the exampel now. There was in particular an error in the class ``PanNet`` in ``mutispectral.py``, the network was not send on the device. [70e6164](https://github.com/deepinv/deepinv/pull/596/commits/70e616488de2b4c67b5bef9d5fca8708386acdc2)

samuro95 commented on examples/basics/demo_tour_mri.py:
```diff
@@ -96,7 +97,7 @@
 dinv.utils.plot(
     {
-        "x": (x := next(iter(DataLoader(knee_dataset)))),
+        "x": (x := next(iter(DataLoader(knee_dataset))).to(device)),
```
why do you add this ? If the device is gpu you do not want to have the image going to gpu for plotting.

samuro95 reviewed (commented):
I added new comments with the new changes.

mh-nguyen712 commented:
We just updated the preview docs for pr. Can your merge with deepinv/main to generate the preview doc @Tmodrzyk ? Thanks!

Tmodrzyk commented:
@mh-nguyen712 It seems I'm up to date with main, I'm not sure about what I should do here ? 
Also, is the CI broken ? I have a ``permission denied`` when building the docs

Andrewwango commented:
I've just merged from main to remove existing changes from this PR

Tmodrzyk commented:
Looking at the docs generated by the CI, it seems the fontsize is back at 17 for some reasons. I'll double check locally.

Andrewwango reviewed (approved):
LGTM, thanks!

jscanvic reviewed (changes requested):
Thanks for the fix!

The font size is really small in the generated docs for the page on 3D diffraction PSFs. Can you check it out?

main:

<img width="1300" height="500" alt="Image" src="https://github.com/user-attachments/assets/f605af9b-57ae-46a0-889b-cf323f35d6bd" />

PR:

<img width="1300" height="500" alt="Image" src="https://github.com/user-attachments/assets/a948a007-d493-4f33-8724-abde98fd5068" />

Tmodrzyk commented:
I changed the CSS to fix Issue #673 causing figures to be plotted side by side instead of one after the other.
I also increased the default font size for the examples and fixed some errors in other plotting functions.
@jscanvic let me know if you spot any other issues in the examples :)

jscanvic reviewed (approved):
LGTM thanks for fixing that!
